tools/ec2recovery.py

```diff
@@ -1,280 +1,211 @@
-        if key == 'Name':
-            value = "%s-restored" % value
-        if key == 'RestoreDate' or 'aws:' in key:
-            continue
-        result.append({
-            'Key': key,
-            'Value': value})
+    if tags:
+        for i in range(len(tags.split(','))):
+            pairs = tags[i].split(':')
+            key = pairs[0]
+            value = pairs[1]
+            result.append({
+                'Key': key,
+                'Value': value})
     result.append({
         'Key': 'RestoreDate',
         'Value': datetime.now().strftime("%Y-%m-%d %H:%M:%S")})
     return result
 
 
-def create_instance(session, dry_run, ami_id, instance, bdm, user_data):
-    # Create volumes from snapshots
-    ec2 = session.resource('ec2')
-    instances = ec2.create_instances(
-        DryRun=dry_run,
-        ImageId=ami_id,
-        MinCount=1,
-        MaxCount=1,
-        KeyName=instance['KeyName'],
-        SecurityGroupIds=pull_sg_ids(instance),
-        InstanceType=instance['InstanceType'],
-        SubnetId=instance['SubnetId'],
-        BlockDeviceMappings=bdm,
-        IamInstanceProfile={'Arn': instance['IamInstanceProfile']['Arn']},
-        UserData=user_data)
-    instance_id = instances[0].instance_id
-    restored = ec2.Instance(instance_id)
-    print "\n\tRestoring instance from snapshot..."
-
-    client = session.client('ec2')
-    waiter = client.get_waiter('instance_running')
-    waiter.wait(
-        DryRun=dry_run,
-        InstanceIds=[instance_id]
-    )
-    restored.create_tags(
-        DryRun=dry_run,
-        Tags=create_tags(instance['Tags']))
-    return instance_id
+def read_user_date(userdata):
+    if not os.path.isfile(userdata):
+        print "Cannot find file '%s'" % userdata
+        return ""
+    with open (userdata, "r") as input:
+        return input.readlines()
 
 
-def confirm_selection(snapshots, ami):
-    while True:
-        clear_screen()
-        print "\tVerify Restore Options\n\n"
-        print "\tAMI ID: %s\n" % ami
-        print "\tSnapshots:"
-        for snap in snapshots:
-            print "\t\tVolumeID: %s, VolumeName: %s, SnapshotID: %s\n" % (
-                snap[0], snap[1], snap[2])
-        response = raw_input("\nAre these values correct? [Y/N]: ")
-        if response.upper() == "Y" or response.upper() == "N":
-            return response
+def create_instance(session, dryrun, ami, keyname,
+                    securitygroups, type, subnet,
+                    snapshots, iamrolename, userdata,
+                    tags):
+    # Create volumes from snapshots
+    ec2 = session.resource('ec2')
+    try:
+        instances = ec2.create_instances(
+            DryRun=dryrun,
+            ImageId=ami,
+            MinCount=1,
+            MaxCount=1,
+            KeyName=keyname,
+            SecurityGroupIds=securitygroups.split(','),
+            InstanceType=type,
+            SubnetId=subnet,
+            BlockDeviceMappings=snapshots,
```

Comment:
we need a little more distinction here, in some cases we are restoring with a partial snapshot set, which i don't think we should fail on, ie. merge snapshots extant against ami block dev, and if perfect union, print a warning.